### DISCUSSION TITLE
How do I add a new currency?

### DISCUSSION THREAD

**Discussion by dtslvr** [ghostfolio/ghostfolio#2881] [Q&A] [answered]

**Comment by dtslvr** [comment]
Go to the _Admin Control_ panel and click the _Add Currency_ button. Then insert e.g. `EUR` in the prompt.

**Reply by deepakpatil23** [reply]
In Admin Control, now Add Currency button is not there.  is that I have to buy premium one?
i have searched at https://localhost:4200/en/admin/settings

**Reply by dtslvr** [reply, marked as answer]
This functionality has been moved. Please find _How do I add a new currency?_ in the [FAQ](https://ghostfol.io/en/faq/self-hosting).

**Reply by nicosemp** [reply]
@dtslvr how can we change the primary currency?

Also how can I delete a currency created by mistake? I can navigate to `/admin/market-data` but all I see is the one asset I added. If I then click the filter dropdown menu I can select "currencies" and they appear. Why is a filter making extra things appear instead of filtering down from a complete list? Anyway I can select the currency I want to delete, but the "Delete" button is disabled.
<img width="225" height="161" alt="image" src="https://github.com/user-attachments/assets/46206c79-2604-405b-8a59-9118887ad589" />

**Reply by nicosemp** [reply]
Found the primary currency control by clicking on the user icon. I'm still confused about how to delete currencies.

**Reply by Section82** [reply]
I also am trying to figure out how to delete currencies…

EDIT: Figured it out! you go to the market data section. Click the filter at the top and then scroll down that list and select "currencies". Once there, select the currency that you would like to delete. Then, **DO NOT** select the three dots on the side of the item as "delete" will be greyed out. Instead, select the three _vertical_ dots at the top and select "Delete profiles"


Does this make sense? No. Is this how you do it? yes -_-

also, you can't delete multiple even though you select multiple. it will just delete one.

**Reply by laszukdawid** [reply]
I'm having an issue with adding a new currency, CAD. I followed the [FAQ](https://ghostfol.io/en/faq/self-hosting) which is:
```
How do I add a new currency?

Ghostfolio manages currencies automatically based on all the recorded activities. If you need an additional currency, you can manually enter it.

    Open the Admin Control panel
    Navigate to the Market Data section
    Click on the + button
    Switch to Add Currency
    Insert e.g. EUR for Euro
    Select Filter by Currencies
    Find the entry USDEUR
    Click the menu item Gather Historical Market Data in the dialog 
```

However, for one, there's no "Filter by Currencies" anywhere on the page, and backend logs show an issue with the `SymbolProfile` [1] which I'm guessing is because the "Symbol Mapping" in the form is just `{}`. It isn't clear to me what should go into the Symbol Mapping. Any suggestion?

[1]
```
An operation failed because it depends on one or more records that were required but not found. No 'SymbolProfile' record (needed to update inlined relation for 'SymbolProfileO verrides' record(s)) was found for a nested upsert on one-to-one relation 'SymbolProfileToSymbolProfileOverrides'.
    at ei.handleRequestError (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:7268)
    at ei.handleAndLogRequestError (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:6593)
    at ei.request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:6300)
    at async a (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:130:9551)
    at async AdminService.patchAssetProfileData (/ghostfolio/apps/api/main.js:1:815084)
    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-execution-context.js:46:28
    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-proxy.js:9:17
    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-explorer.js:199:17 {
  code: 'P2025',
  meta: {
    modelName: 'SymbolProfile',
    cause: "No 'SymbolProfile' record (needed to update inlined relation for 'SymbolProfileOverrides' record(s)) was found for a nested upsert on one-to-one relation 'SymbolProlProfileOverrides'."
  },
  clientVersion: '6.19.0'
}
```

**Reply by anthortic** [reply]
`
[Nest] 1  - 02/20/2026, 11:03:45 AM     LOG [DataGatheringProcessor (GATHER_HISTORICAL_MARKET_DATA)] Historical market data gathering has been completed for USDEUR (YAHOO) at 2026-02-20

[Nest] 1  - 02/20/2026, 11:04:05 AM   ERROR [ExceptionsHandler] PrismaClientKnownRequestError: 

Invalid `prisma.symbolProfile.update()` invocation:

An operation failed because it depends on one or more records that were required but not found. No 'SymbolProfile' record (needed to update inlined relation for 'SymbolProfileOverrides' record(s)) was found for a nested upsert on one-to-one relation 'SymbolProfileToSymbolProfileOverrides'.

    at ei.handleRequestError (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:7268)

    at ei.handleAndLogRequestError (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:6593)

    at ei.request (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:121:6300)

    at async a (/ghostfolio/apps/api/node_modules/@prisma/client/runtime/library.js:130:9551)

    at async AdminService.patchAssetProfileData (/ghostfolio/apps/api/main.js:1:817210)

    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-execution-context.js:46:28

    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-proxy.js:9:17

    at async /ghostfolio/apps/api/node_modules/@nestjs/core/router/router-explorer.js:199:17 {

  code: 'P2025',



{

    modelName: 'SymbolProfile',

    cause: "No 'SymbolProfile' record (needed to update inlined relation for 'SymbolProfileOverrides' record(s)) was found for a nested upsert on one-to-one relation 'SymbolProfileToSymbolProfileOverrides'."

  },

  clientVersion: '6.19.0'

}
`

Same issue here.
Are you self-hosting?

**Reply by laszukdawid** [reply]
Yes, I'm self-hosting
